plugins/modules/yum_versionlock.py

```diff
@@ -122,7 +121,15 @@ def match(entry, name):
         m = NEVRA_RE_DNF.match(entry)
     if not m:
         return False
-    return fnmatch(m.group("name"), name)
+
+    # Validate if the given package name matches
+    # This will validate:
+    #   - package name like:             httpd
+    #   - package name and version like: httpd-0:2.4.57-3.el9
+    if name in m.group():
+        return True
+    else:
+        return False
```

Comment:
1. Why not simply `return name in m.group()`? That's equivalent to the above four lines.
2. If the package name is `ttp`, it will also match `httpd`. That looks wrong to me.
3. I think `fnmatch` is used to also match wildcards (see the documentation for `name`, it explicitly mentions them). The new code doesn't handle wildcards.

Reply:
Ok, we can keep the fnmatch solution. Will push later.